### DISCUSSION TITLE
How to convert an NDJSON stream to a YAML stream?

### DISCUSSION THREAD

**Discussion by anatoly-scherbakov** [mikefarah/yq#1279] [Q&A]
Thank you for the great tool!

I am trying to convert an [NDJSON](http://ndjson.org/) file (also known as [JSON Sequence](https://datatracker.ietf.org/doc/html/rfc7464)) with the following content to YAML:

```json
{"foo": "bar"}
{"baz": "kaboom"}
```

I expect to see:

```yaml
foo: bar
---
baz: kaboom
```

`yq` doesn't accept `sample.ndjson` as a valid input:

```bash
$ yq . sample.ndjson 
{"foo": "bar"}
Error: bad file 'sample.ndjson': yaml: line 1: did not find expected <document start>
```

`yq` treats JSON as a subset of YAML, but NDJSON isn't such a subset, hence the error. Is there any known compact way to achieve this conversion besides processing each line separately using `xargs` and then inserting document separators? Thank you!

**Comment by mikefarah** [comment]
At the moment, that's the only way (using xargs). 

I've just started looking into json decoders - and found that go's built in decoder supports NDJSON! 

I'll start working on adding a specific json decoder, once it's in you'll need to specify the decoder with a `-p` flag :)

**Comment by mikefarah** [comment]
I've added the decoder in v4.27.1 :) Just use `-p=json` when reading the file
